### ISSUE TITLE
Passing i_handle.s through ic960 then asm960?

### ISSUE THREAD

**Issue by biggestsonicfan** [cglmrfreeman/i960-CTOOLS-with-NINDY#2]
In the Windows version of QTMAKE.bat, `i_handle.s` is passed like this:
`ic960 -E i_handle.c | asm960 -o i_handle.o`

This is an excerpt from the NINDY User Manual:
> There are a few assembly files which have pre-processor directives in them and need special consideration. These will not run straight through the compiler and come out with the correct results. "i_handle.s" is one of those files and must be re-compiled as follows:
> ```
> copy i_handle.s i_handle.c
> ic960 -E i_handle.c | asm960 -o i_handle.o
> del i_handle.c
> ```
>The assembly file must be copied to a .c file extension to run through the pre-processing phase of the C compiler. When we are done, the C version of the file is removed. The -E option on the compiler sends the file through the C pre-processor and ouputs to standard output. It is then piped to the assembler for assembly.

It was assumed that the following command would work:
```
cp i_handle.s i_handle.c
I960BASE=~/i960-CTOOLS-with-NINDY/i386-nbsd1-ctools/ ../i386-nbsd1-ctools/bin/ic960 -E i_handle.c | I960BASE=~/i960-CTOOLS-with-NINDY/i386-nbsd1-ctools/ ../i386-nbsd1-ctools/bin/asm960 -o i_handle.o
rm i_handle.c
```
However, that results in strange errors:

> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: 
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> ../i386-nbsd1-ctools/bin/asm960: .globl: not found
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr1:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr2:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr3:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr4:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr5:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr6:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr7:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr8:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr9:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr10:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr11:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr12:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr13:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr14:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr15:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr16:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr17:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr18:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr19:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr20:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr21:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr22:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr23:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr24:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr25:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr26:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr27:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr28:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr29:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr30:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_intr31:
> : not foundd1-ctools/bin/asm960: ret
> : not foundd1-ctools/bin/asm960: 
> : not foundd1-ctools/bin/asm960: _user_NMI:
> : not foundd1-ctools/bin/asm960: 
> ../i386-nbsd1-ctools/bin/asm960: ldconst: not found
> ../i386-nbsd1-ctools/bin/asm960: addo: not found
> : not foundd1-ctools/bin/asm960: 
> ../i386-nbsd1-ctools/bin/asm960: 253: Syntax error: "(" unexpected

It's almost as if the pre-processor is not handling the file correctly? Unsure how to proceed.

EDIT1: It looks like the pre-processor is doing is just stripping away all commented lines? I will attempt this manually in a test file and pull it into my VM and try to assemble that.

**Comment by biggestsonicfan** [comment]
I rearranged the i_handle.s a bit. First I removed all commented lines. Then I removed all debug "if" cases, as they were stripped away in the pre-processor anyway. Then I arranged the `.globl` to how they appear in the assembler guide.
Instead of
```
.glob a
.glob b
.glob c
a
ret
b
ret
c
ret
```
I organized them like so:
```
.glob a
a
ret
.glob b
b
ret
.glob c
c
ret
```

After passing the new i_handle.c to the pre-processor and outputting (>) to i_handle_test.c then passing i_handle_test.c to the assembler to output i_handle.o, it worked. Will keep an eye on if this affects anything or not, but I am closing this as fixed from commit 9e967b524278724dacb46f808f09e23bccd42f3f.
